contracts/src/chainlink/solidity/emergency/StarkNetValidator.sol

```diff
@@ -74,15 +122,152 @@ contract StarkNetValidator is TypeAndVersionInterface, AggregatorValidatorInterf
     uint256, /* currentRoundId */
     int256 currentAnswer
   ) external override checkAccess returns (bool) {
-    bool status = currentAnswer == ANSWER_SEQ_OFFLINE;
-    uint256[] memory payload = new uint256[](2);
+    return _sendUpdateMessageToL2(currentAnswer);
+  }
+
+  /**
+   * @notice retries to send the latest answer as update message to L2
+   * @dev only with access, useful in cases where a previous x-domain message was handeled unsuccessfully.
+   */
+  function retry() external checkAccess returns (bool) {
+    (, int256 latestAnswer, , , ) = AggregatorV3Interface(s_source).latestRoundData();
+    return _sendUpdateMessageToL2(latestAnswer);
+  }
+
+  /// @notice sends the 'update_status(answer, timestamp)' message via the L1 -> L2 bridge to the L2 feed contract
+  function _sendUpdateMessageToL2(int256 answer) internal returns (bool) {
+    // Bridge fees are paid on L1
+    uint256 fee = _approximateFee();
 
     // Fill payload with `status` and `timestamp`
+    uint256[] memory payload = new uint256[](2);
+    bool status = answer == ANSWER_SEQ_OFFLINE;
     payload[0] = toUInt256(status);
     payload[1] = block.timestamp;
+
     // Make the StarkNet x-domain call.
-    // NOTICE: we ignore the output of this call (msgHash, nonce), and we don't raise any events as the event LogMessageToL2 will be emitted from the messaging contract
-    STARKNET_CROSS_DOMAIN_MESSENGER.sendMessageToL2(L2_UPTIME_FEED_ADDR, STARK_SELECTOR_UPDATE_STATUS, payload);
+    // NOTICE: we ignore the output of this call (msgHash, nonce).
+    // We also don't raise any events as the 'LogMessageToL2' event will be emitted from the messaging contract.
+    STARKNET_CROSS_DOMAIN_MESSENGER.sendMessageToL2{value: fee}(
+      L2_UPTIME_FEED_ADDR,
+      SELECTOR_STARK_UPDATE_STATUS,
+      payload
+    );
     return true;
   }
+
+  /// @notice L1 oracle is asked for a fast L1 gas price, and the price multiplied by the configured gas estimate
+  function _approximateFee() internal view returns (uint256) {
+    (, int256 fastGasPriceInWei, , , ) = AggregatorV3Interface(s_gasConfig.gasPriceL1Feed).latestRoundData();
+    return (uint256(fastGasPriceInWei) * s_gasConfig.gasUsed * s_gasConfig.buffer);
```

Comment:
@cds95 I've removed the `/ 1e18` multiplier here. Shouldn't we send the fee in wei?